zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/Commands.java

```diff
@@ -62,13 +64,17 @@
     /** Maps command names to Command instances */
     private static Map<String, Command> commands = new HashMap<String, Command>();
     private static Set<String> primaryNames = new HashSet<String>();
+    private static Set<String> commandsAvailableBeforePeerSync = new HashSet<String>();
 
     /**
      * Registers the given command. Registered commands can be run by passing
      * any of their names to runCommand.
      */
-    public static void registerCommand(Command command) {
+    public static void registerCommand(Command command, boolean enabledBeforePeerSync) {
```

Comment:
nit: You can overload the constructor to avoid modifying the signature at so many places.